src/libraries/System.Text.Json/tests/Serialization/ReferenceHandlingTests.Deserialize.cs

```diff
@@ -1118,7 +1122,7 @@ public static void ReferenceObjectsShouldNotContainMoreProperties()
             }";
 
             ex = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<Employee>(json, s_deserializerOptionsPreserve));
-            Assert.Equal("$.Manager", ex.Path);
+            Assert.Equal("$.Manager.$ref", ex.Path);
```

Comment:
The Json path shouldn't be `$.Manager.Name` in this case? The payload was fine until it hit the trailing property `Name` in the reference object.

Reply:
This is a cost:benefit area. The exception says "A JSON object that contains a '$ref' metadata property must not contain any other properties.". The options as I see them:
1) Make the path `Name` (not `$ref`). However to do that means we need to read and look up the "invalid" property name which we don't do elsewhere in cases like this and there are some edge cases like the property may be missing on the object side which would complicate showing the "invalid" property name.
2) Throw without $ref in the path. This seems to be the most "correct" easy solution. I'll change it to this.

Reply:
Based on offline discussion, we implemented option 1 above.